File: VB/WebSite/App_Code/CustomMenuViewCallbackCommand.vb
Imports System
Imports System.Data
Imports System.Web
Imports System.Web.UI
Imports DevExpress.Web
Imports DevExpress.Web.ASPxScheduler
Imports DevExpress.Web.ASPxScheduler.Internal

Public Class CustomMenuViewCallbackCommand
    Inherits MenuViewCallbackCommand

    Private currentCommandProhibited As Boolean

    Public Sub New(ByVal control As ASPxScheduler)
        MyBase.New(control)
    End Sub

    Protected Overrides Sub ParseParameters(ByVal parameters As String)
        Dim isNewAppointmentCommand As Boolean = (parameters = "NewAppointment" OrElse parameters = "NewAllDayEvent" OrElse parameters = "NewRecurringAppointment" OrElse parameters = "NewRecurringEvent")

        currentCommandProhibited = isNewAppointmentCommand AndAlso Not ResourcesAvailabilities.IsIntervalAvailableForResource(Control.SelectedResource.Id.ToString(), Control.SelectedInterval)

        MyBase.ParseParameters(parameters)
    End Sub

    Protected Overrides Sub ExecuteCore()
        If currentCommandProhibited Then
            AddHandler Control.CustomJSProperties, AddressOf ASPxScheduler1_CustomJSProperties
        Else
            MyBase.ExecuteCore()
        End If
    End Sub

    Private Sub ASPxScheduler1_CustomJSProperties(ByVal sender As Object, ByVal e As CustomJSPropertiesEventArgs)
        e.Properties.Add("cpWarning", ResourcesAvailabilities.WarningMessage)
    End Sub
End Class

File: CS/WebSite/App_Code/CustomMenuViewCallbackCommand.cs
using System;
using System.Data;
using System.Web;
using System.Web.UI;
using DevExpress.Web.ASPxClasses;
using DevExpress.Web.ASPxScheduler;
using DevExpress.Web.ASPxScheduler.Internal;

public class CustomMenuViewCallbackCommand : MenuViewCallbackCommand {
    bool currentCommandProhibited;

    public CustomMenuViewCallbackCommand(ASPxScheduler control)
        : base(control) {
    }

    protected override void ParseParameters(string parameters) {
        bool isNewAppointmentCommand = (parameters == "NewAppointment" ||
            parameters == "NewAllDayEvent" ||
            parameters == "NewRecurringAppointment" ||
            parameters == "NewRecurringEvent");

        currentCommandProhibited = isNewAppointmentCommand && !ResourcesAvailabilities.IsIntervalAvailableForResource(
            Control.SelectedResource.Id.ToString(), Control.SelectedInterval);

        base.ParseParameters(parameters);
    }

    protected override void ExecuteCore() {
        if (currentCommandProhibited) {
            Control.CustomJSProperties += new CustomJSPropertiesEventHandler(ASPxScheduler1_CustomJSProperties);
        }
        else {
            base.ExecuteCore();
        }
    }

    void ASPxScheduler1_CustomJSProperties(object sender, CustomJSPropertiesEventArgs e) {
        e.Properties.Add("cpWarning", ResourcesAvailabilities.WarningMessage);
    }
}